### PR TITLE
Update when FAB should close or hide while navigating the app

### LINKED ISSUE #10797: [Bug] The FAB does not disappear when searching decks
###### Reproduction Steps

1. Tap search button (appears when you have at least 10 decks)
2. Close keyboard that pops up

###### Expected Result
The floating action button (FAB) should disappear when a search is initiated.

###### Actual Result
The FAB moves up as a small dot that is still actionable and results in a weird issue when tapped upon. It's too cramped to still appear, and when tapping on a selection, only some of the options appear depending on the screen size. If the FAB is tapped, backing out takes an extra step (3 back button taps) to exit out of the search.

On the most recent alpha, v 2.16alpha56, the FAB does not shrink in size for me, but that may be related to my emulator. Regardless, the issue still persists.

###### Debug info
```
AnkiDroid Version = 2.15.6

Android Version = 12

Manufacturer = samsung

Model = SM-S901U

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 12; SM-S901U Build/SP1A.210812.016; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/99.0.4844.88 Mobile Safari/537.36

ACRA UUID = 6513c4b2-d751-40e6-9fab-f42c1fc1e7f9

Scheduler = std2

Crash Reports Enabled = false

DatabaseV2 Enabled = true
```

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

https://user-images.githubusercontent.com/71997294/163003201-51d23947-fe74-42c3-9a87-4475d68e208a.mp4

v 2.15.6

https://user-images.githubusercontent.com/71997294/163004244-3f240420-abc2-4c82-b7c6-8a12138a50eb.mp4

v 2.16alpha56

### PR DESCRIPTION
**Pull request by winstoncooke**
## Purpose / Description
When pulling up the keyboard to search, the FAB did not disappear, which resulted in unexpected behavior when interacted with.

## Fixes
Fix #10797 

## Approach
I set the fab to disappear when the SearchItem toolbar is opened and reappear when it is closed.

## How Has This Been Tested?

This was manually tested on the latest alpha using the emulator for Pixel 5 API 31.
I looked into a unit test, but I faced the following issue: https://stackoverflow.com/questions/36793404/checking-android-fab-visibility-always-returns-visible-from-robolectric-test-cas

## Learning
This SO post can be applied to converting the code to kotlin when the time comes: https://stackoverflow.com/questions/57712498/how-to-hide-fab-when-user-click-on-searchview-and-keyboard-opens

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

https://user-images.githubusercontent.com/71997294/163465158-6dd476f0-8837-46c2-b820-628e62b18609.mp4

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 1]
```diff
@@ -960,6 +982,11 @@ private void automaticSync() {
     @Override
     @KotlinCleanup("once in Kotlin: use HandlerUtils.executeFunctionWithDelay")
     public void onBackPressed() {
+        // If floating action button is not visible, make it visible
+        if (!mFloatingActionMenu.isFabVisible()) {
+            mFloatingActionMenu.showFloatingActionButton();
+        }
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:988]: I'm not sure I understand when this occurs, is `onMenuItemActionCollapse` not sufficient in this case?

Comment (winstoncooke) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java]: Great catch. I didn't hit submit on my final commit that removed that code. It's removed now.

**Review by david-allison — APPROVED** [review]
I'd be tempted to move the 'if hidden' + logging logic inside `mFloatingActionMenu` for:

closeFloatingActionMenu | hideFab | showFAB

But, this looks good. Implementer's choice

**Comment by winstoncooke** [comment]
@david-allison I think that's a good idea. I'll move the logic into `DeckPickerFloatingActionMenu.kt`.

**Comment by winstoncooke** [comment]
I moved the logic into `DeckPickerFloatingActionMenu.kt` and also made changes so that the floating action menu apropriately collapses when it's expanded when clicking on action bar items and the hamburger menu.

**Review by mikehardy — APPROVED** [review]
Nice, with the move to the FAB code - DickPicker is big enough :-)
Thanks for this!

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 2]
```diff
@@ -278,7 +278,6 @@ public class DeckPicker extends NavigationDrawerActivity implements
     private void onDeckClick(View v, DeckSelectionType selectionType) {
         long deckId = (long) v.getTag();
         Timber.i("DeckPicker:: Selected deck with id %d", deckId);
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:280]: q: why are we no longer closing the menu on this case?

Comment (winstoncooke) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:280]: You can't actually click on a deck when the floating action menu is expanded. Clicking on the screen where the deck is will close the floating action menu, meaning the code never executes and is redundant as far as I can tell.

**Comment by david-allison** [comment]
Emulator flaky, haven't seen this one in a long time

```
com.ichi2.anki.tests.ContentProviderTest > testInsertAndUpdateModel[0][TestingAVD(AVD) - 11] FAILED 
	java.lang.RuntimeException: android.os.DeadSystemException
	at android.os.storage.StorageManager.getVolumeList(StorageManager.java:1383)
```

**Comment by winstoncooke** [comment]
@david-allison that's not a result from my end is it?

**Comment by mikehardy** [comment]
Boom! Thanks

**Pull request merged**